plugins/core/IO/qLASIO/include/CopcVlrs.h

```diff
@@ -0,0 +1,270 @@
+// #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         #
+// #  GNU General Public License for more details.                          #
+// #                                                                        #
+// #                   COPYRIGHT: Hobu, Inc.                           	    #
+// #                                                                        #
+// ##########################################################################
+
+// qLASIO
+#include "LasDetails.h"
+
+// CCCoreLib
+#include <CCGeom.h>
+
+// Qt
+#include <QDataStream>
+
+// System
+#include <array>
+#include <cstddef>
+#include <cstdint>
+
+/// COPC structs as defined in the specs
+/// We kept COPC naming convention (snake case)
+/// this is based on COPC v1 specs. See https://copc.io for documentation
+namespace copc
+{
+	struct Info
+	{
+		/// Extract extent
+		///
+		/// return true if extent is valid false otherwise
+		/// Note that COPC extent is cubic, it's the extent of the
+		/// COPC octree, not the BB of the points.
+		bool extractExtent(LasDetails::UnscaledExtent& extent) const
+		{
+			extent.clear();
+			CCVector3d center(center_x, center_y, center_z);
+			CCVector3d halfsizeVec(halfsize, halfsize, halfsize);
+			extent.add(center + halfsizeVec);
+			extent.add(center - halfsizeVec);
+			// it should be valid ;) but we test its norm too
+			// maybe we could also test it's a cube
+			return extent.isValid() && extent.getDiagNormd() > 0;
+		}
+
+		/// Overload the stream extraction operation to read an Info object from a QDataStream.
+		friend QDataStream& operator>>(QDataStream& stream, Info& copc_info)
+		{
+			// use quint64 for linux compat (macOS and Win are fine)
+			quint64 root_hier_offset_, root_hier_size_, reserved_;
+			stream.setByteOrder(QDataStream::ByteOrder::LittleEndian);
+			stream >> copc_info.center_x >> copc_info.center_y >> copc_info.center_z;
+			stream >> copc_info.halfsize >> copc_info.spacing;
+			stream >> root_hier_offset_ >> root_hier_size_;
+			copc_info.root_hier_offset = root_hier_offset_;
+			copc_info.root_hier_size   = root_hier_size_;
+			stream >> copc_info.gpstime_minimum >> copc_info.gpstime_minimum;
+			// Flush reserved
+			std::for_each(std::begin(copc_info.reserved), std::end(copc_info.reserved), [&stream, &reserved_](auto& reserved)
+			              { stream >> reserved_; });
+			return stream;
+		};
+
+		static constexpr size_t SIZE = 160;
+
+		/// Geometric parameters (root cell extent)
+		double center_x{0.0};
+		double center_y{0.0};
+		double center_z{0.0};
+		double halfsize{0.0};
+		/// minimum distance between points at root (0) level.
+		/// it is supposed to be halved at each level
+		double spacing{0.0};
+
+		/// Byte offeset in the LAS file
+		/// the root node is not necessary the node comming at first
+		/// in the COPC EVRL, so it's not the start of the COPC
+		/// EVRL Reccord.
+		uint64_t root_hier_offset{0};
+		uint64_t root_hier_size{0};
+
+		/// TODO
```

Comment:
TODO?

Reply:
values are unpacked from the file but it's not used anywhere. I wonder if there is a specific use cases for these values since If I understand correctly we compute our own shift in CC.